File: Day336/WidestVerticalAreaBetweenTwoPointsContainingNoPoints.go
func maxWidthOfVerticalArea(points [][]int) int {
    sort.Slice(points, func (i, j int) bool {
        if points[i][0] == points[j][0] {
            return points[i][1] < points[j][1]
        }
        return points[i][0] < points[j][0]
    })
    answer := math.MinInt
    for i:=1; i<len(points); i++ {
        answer = int(math.Max(float64(answer), float64(points[i][0] - points[i-1][0])))
    }
    return answer
}

File: Day336/WidestVerticalAreaBetweenTwoPointsContainingNoPoints.java
class Solution {
    public int maxWidthOfVerticalArea(int[][] points) {
        Arrays.sort(points, (a,b)->{
            if (a[0] == b[0]) {
                return a[1] - b[1];
            }
            return a[0]-b[0];
        });
        int answer = Integer.MIN_VALUE;
        for (int i=1; i<points.length; i++) {
            answer = Math.max(answer, points[i][0] - points[i-1][0]);
        }
        return answer;
    }
}